src/models/player.rs

```diff
@@ -664,12 +668,18 @@ fn next_video_update(
                     .unwrap_or_default();
                 next_season != 0 || current_season == next_season
             })
-            .map(|(_, next_video)| {
-                let mut next_video = next_video.clone();
-                if let Some(stream) = stream {
-                    next_video.streams = vec![stream.clone()];
+            .and_then(|(_, next_video)| {
+                match next_video.released {
```

Comment:
Since this released property is not exactly accurate I'm not entirely sure whether it's a good idea not allow going to the next video based on it. Since I'd say it's best to see whether are are valid stream or not for the user. But don't have a strong opinion 🤷